# Content Index
- [00.Basic information & opreation](#00)
  - [00-00.Ubuntu location in Windows10](#00-00)
# Set up your Ubuntu OS on windows for RNA-seq

<h2 id="00">00-00.Ubuntu location in Windows10</h2>

<h3 id="00-00">00-00.Ubuntu location in Windows10</h3>

File location for Ubuntu in Windows10
```
%userprofile%\AppData\Local\Packages
```
A folder cotains "Ubuntu", looks like `CanonicalGroupLimited.Ubuntu18.04onWindows_79rhkp1fndgsc`
### 00-01.Increase resource limits on Ubuntu
#### Edit `/etc/security/limits.conf`
```bash
$ cd /etc/security
$ sudo nano limits.conf
```
Then add belows in this file
```
\*          soft   nofile   16384
\*          hard   nofile   16384
root       soft   nofile   16384
root       hard   nofile   16384
username   soft   nofile   16384
username   hard   nofile   16384
```
#### Edit '/etc/pam.d/common-session add'
```bash
$ cd /etc/pam.d
$ sudo nano common-session*
```
Then add belows in this file
```
session required pam_limits.so
```
And command below after edit
```bash
$ ulimit -n 16384
```
### 00-02.Basic command in Linux (Ubuntu)
#### Add specific usr into sudo folder
```bash
adduser username
sudo usermod -aG sudo username
```

#### Login specific user
```bash
$ su - uekilab
```
#### Number of files with certain extension
```bash
$ ls | grep '\.fastq$' | wc -l
```
#### List of files with certain extension
```bash
$ ls | grep '\.fastq$'
```
#### Return nth of files with certain extension
```bash
$ vim "$(find . -maxdepth 1 -type f  | awk 'NR==12')"
$ vim "$(ls -1 | awk 'NR==12')"
```
#### Check the file of environment virables
```bash
$ nano ~/.bashrc
```
#### Get number of CPU core
```bash
$ cat /proc/cpuinfo | grep processor | wc -l
$ grep -c ^processor /proc/cpuinfo
```
#### Unzip compressed files
```bash
$ tar -xvzf
$ gunzip
```
#### Change configure file permission
```bash
$ chmod +x configure
```
#### Check first 5 lines of fastq file
```bash
(zcat for gz compressed file)
$cat Sample_R1.fastq | sed 's/@//g;s/ /_/g' | awk '{if(NR%4==1)print ">"$0;if(NR%4==2) print $0;}' | head
$cat Sample_R2.fastq | sed 's/@//g;s/ /_/g' | awk '{if(NR%4==1)print ">"$0;if(NR%4==2) print $0;}' | head
```
#### download convert tool from [UCSC](http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bigBedToBed)
```bash
$ chmod +x bigBedToBed
```
#### download bigBed file from [Ensembl](ftp://ftp.ensembl.org/pub/release-97/bed/)
```bash
$ bigBedToBed gerp_constrained_elements.mus_musculus.bb mus_musculus.bed
$ sed -e 's/^\([0-9XY]\)/chr\1/' -e 's/MT/chrM/' mus_musculus.bed > mus_musculus.chr.bed
```
## 01.Installation of softwares for RNA-seq analysis
### 01-00.user login then enter into root user
```bash
$ su - icmh
$ sudo -s
$ cd /home
$ mkdir InstallationPackages
$ cd /InstallationPackages
````
### 01-01.Installation of Bioconda
#### Download 64-bit python 3.7 from [here](https://docs.conda.io/en/latest/miniconda.html)
```bash
$ wget -P ./ "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh"
$ sudo chmod +x ./Miniconda3-latest-Linux-x86_64.sh
$ ./Miniconda3-latest-Linux-x86_64.sh
$ echo 'export PATH=/home/icmh/miniconda3/bin:$PATH' >> ~/.bashrc
$ source ~/.bashrc
```
#### Add the bioconda channel:
```bash
$ conda config --add channels defaults
$ conda config --add channels bioconda
$ conda config --add channels conda-forge
```
#### Create a dDocent conda environment:
```bash
$ conda create -n ddocent_env ddocent
```
#### Switching between Python 2 and Python 3 environment
```bash
$ conda create --name py2 python=2.7
$ conda create --name py3 python=3.5
$ conda activate py2
$ conda deactivate
```
### 01-02.Installation of Salmon
#### [Master repository of Salmon in GitHub](https://combine-lab.github.io/salmon/getting_started/)
```bash
$ conda config --add channels conda-forge
$ conda config --add channels bioconda
$ conda create -n salmon salmon
$ conda activate salmon
$ conda deactivate
```
#### Download salmon tools for generation of decoy index
```bash
$ sudo apt-get install bedtools
$ sudo apt-get install autoconf
$ sudo apt-get install zlib1g-dev
$ cd /home/InstallationPackages
$ git clone https://github.com/marbl/MashMap
$ cd MashMap
$ ./bootstrap.sh
$ ./configure
$ make -j $(grep -c ^processor /proc/cpuinfo)
$ echo 'export PATH=/home/InstallationPackages/MashMap:$PATH' >> ~/.bashrc
$ source ~/.bashrc
$ cd ~
$ wget -P ./ "https://raw.githubusercontent.com/COMBINE-lab/SalmonTools/master/scripts/generateDecoyTranscriptome.sh"
$ chmod +x generateDecoyTranscriptome.sh
```
#### Salmon example code for index building with decoy
##### resource download
* FASTa (Transcriptome; Date Modified: 5/24/19):  
ftp://ftp.ensembl.org/pub/release-97/fasta/mus_musculus/cdna/Mus_musculus.GRCm38.cdna.all.fa.gz
* FASTa (Genome; Date Modified: 5/24/19):  
Ftp://ftp.ensembl.org/pub/release-97/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.toplevel.fa.gz
* GTF (Date Modified: 5/26/19):  
ftp://ftp.ensembl.org/pub/release-97//gtf/mus_musculus/Mus_musculus.GRCm38.97.gtf.gz
#### build a decoy-aware transcriptome file and do quantification using selective alignment
```bash
$ ~/generateDecoyTranscriptome.sh -j $(grep -c ^processor /proc/cpuinfo) -g /mnt/d/mm10_FASTa/Mus_musculus.GRCm38.dna.toplevel.fa -t /mnt/d/mm10_FASTa/Mus_musculus.GRCm38.cdna.all.fa -a /mnt/d/mm10_GTF/Mus_musculus.GRCm38.97.gtf -o TranscriptIndex_ENSEMBL_Salmon_Decoy
$ salmon index -t /mnt/d/TranscriptIndex_ENSEMBL_Salmon_Decoy/gentrome.fa -i TranscriptIndex_ENSEMBL_Salmon_Index -d /mnt/d/TranscriptIndex_ENSEMBL_Salmon_Decoy/decoys.txt -k 31
```
#### build a normal index
```bash
$ salmon index -t Mus_musculus.GRCm38.cdna.all.fa.gz -i GenomeIndex_ENSEMBL_Salmon_TranscriptFASTa
```
Please check [here](https://combine-lab.github.io/salmon/getting_started/) for more detials.
### 01-03.Installation of STAR
#### Compilling from [source](https://github.com/alexdobin/STAR)
```bash
$ sudo apt-get update
$ sudo apt-get install g++
$ sudo apt-get install make
$ sudo apt-get install make-guile
$ sudo apt install zlib1g-dev
$ git clone https://github.com/alexdobin/STAR.git
$ cd STAR/source
$ make -j $(grep -c ^processor /proc/cpuinfo) STAR
$ echo 'export PATH=/home/InstallationPackages/STAR/source:$PATH' >> ~/.bashrc
$ source ~/.bashrc
$ STAR --version
```
#### Resource download
* (Reference genome) Downloaded from Ensembl  
https://www.ensembl.org/index.html
* FASTa (Date Modified: 5/24/19):  
ftp://ftp.ensembl.org/pub/release-97/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.primary_assembly.fa.gz
* FASTa Transcriptome (Date Modified: 5/25/19):  
ftp://ftp.ensembl.org/pub/release-97/fasta/mus_musculus/cdna/Mus_musculus.GRCm38.cdna.all.fa.gz
* GTF (Date Modified: 5/26/19):  
ftp://ftp.ensembl.org/pub/release-97/gtf/mus_musculus/Mus_musculus.GRCm38.97.chr.gtf.gz
#### STAR code example for index building
```bash
$ STAR --runThreadN -j $(grep -c ^processor /proc/cpuinfo) --runMode genomeGenerate --genomeDir /mnt/d/GenomeIndex_ENSEMBL_STAR_ReadLength99/ --genomeFastaFiles /mnt/d/mm10_FASTa/Mus_musculus.GRCm38.dna.primary_assembly.fa --sjdbGTFfile /mnt/d/mm10_FASTa/Mus_musculus.GRCm38.97.chr.gtf --sjdbOverhang 99
```
#### STAR code example for mapping
```bash
$ STAR --runThreadN -j $(grep -c ^processor /proc/cpuinfo) --alignEndsType EndToEnd --outSAMtype BAM SortedByCoordinate --outSAMmapqUnique 60 --outFileNamePrefix ./Sample_1. --genomeDir ./GenomeIndex_ENSEMBL_STAR_ReadLength49 --readFilesIn ./Read_1.fastq ./Read_2.fastq
```
### 01-04.Installation of rMATS
[Offical website](http://rnaseq-mats.sourceforge.net/rmatsdockerbeta/index.html)
```bash
$ cd /home/InstallationPackages
$ tar -xzf rMATS.4.0.2.tgz
$ sudo apt-get update
$ conda activate py2
$ pip install numpy
$ sudo apt-get install libblas-dev liblapack-dev
$ sudo apt-get install libgsl0-dev
$ sudo apt-get install gfortran
$ sudo apt-get install libgfortran3
```
#### Download GSL from [here](https://www.gnu.org/software/gsl/)
```bash
$ wget -P ./ "ftp://ftp.gnu.org/gnu/gsl/gsl-latest.tar.gz"
$ tar -xzf gsl-latest.tar.gz
```
#### Guide for installation of GSL in [here](https://www.gnu.org/software/gsl/doc/html/intro.html#obtaining-gsl) and [here](https://coral.ise.lehigh.edu/jild13/2016/07/11/hello/)
```bash
$ cd gsl-2.5
$ chmod +x configure
$ ./configure --prefix=/home/InstallationPackages/gsl-2.5
$ make -j $(grep -c ^processor /proc/cpuinfo)
$ make -j $(grep -c ^processor /proc/cpuinfo) check
$ make -j $(grep -c ^processor /proc/cpuinfo) install
```
#### Guide of setting Linux library path in [here](https://www.cyberciti.biz/faq/linux-setting-changing-library-path/) and [here](https://www.tecmint.com/understanding-shared-libraries-in-linux/), Guide for using vi in [here](https://www.howtogeek.com/102468/a-beginners-guide-to-editing-text-files-with-vi/)
#### Check current library path
```bash
$ mv /home/InstallationPackages/gsl-2.5/lib /usr/lib
$ vi /etc/ld.so.conf.d/libgsl.conf
  /usr/lib/lib
$ ldconfig -n /usr/lib/lib
$ ldconfig
$ ldconfig -v | less
```
#### Make soft link to libgsl.so.0
```bash
$ sudo -s
$ ln -s /usr/lib/lib/libgsl.so /usr/lib/lib/libgsl.so.0
$ sudo ldconfig
```
#### Check Unicode type of Python
```bash
$ conda activate py2
$ python
>>> import sys
>>> print sys.maxunicode
1114111
65535
>>> exit()
```
This, '1114111', output indicates that your python is built with --enable-unicode=ucs4, and you should use rMATS-turbo-xxx-UCS4.  
This, '65535', output indicates that your python is built with --enable-unicode=ucs2, and you should use rMATS-turbo-xxx-UCS2.
#### Testing
```bash
$ mkdir /home/InstallationPackages/testData
$ tar -zxf testData.tgz --directory /home/InstallationPackages/testData/
$ wget -P /home/InstallationPackages/testData "http://labshare.cshl.edu/shares/gingeraslab/www-data/dobin/STAR/STARgenomes/ENSEMBL/homo_sapiens/ENSEMBL.homo_sapiens.release-75/Homo_sapiens.GRCh37.75.gtf"
$ cd /home/InstallationPackages/testData
$ python /home/InstallationPackages/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rmats.py --b1 b1.txt --b2 b2.txt --gtf Homo_sapiens.GRCh37.75.chr.gtf --od bam_test -t paired --readLength 50 --cstat 0.0001 --libType fr-unstranded --nthread $(grep -c ^processor /proc/cpuinfo)
```
### 01-05.Installation of Bowtie2
```bash
$ conda activate
$ conda install bowtie2
```
### 01-06.Installation of tophat2
```bash
$ sudo apt-get install tophat
$ sudo -s
$ conda activate py3
$ pip install Tophat
$ conda activate py2
$ pip install Tophat
```
### 01-07.Installation of RSeQC
```bash
$ conda activate py3
$ pip install RSeQC
```
#### Download Perl script gtf2bed from [here](https://expressionanalysis.github.io/ea-utils/)
```bash
$ cd /home/InstallationPackages
$ git clone https://github.com/ExpressionAnalysis/ea-utils
$ cd ea-utils
$ sudo mv clipper /home
$ cd /home/clipper/sparsehash-2.0.3
$ chmod +x configure
$ cd /home/clipper
$ make
$ perl /home/clipper/gtf2bed Mus_musculus.GRCm38.97.chr.gtf > mus_musculus.bed
$ sed -e 's/^\([0-9XY]\)/chr\1/' -e 's/MT/chrM/' mus_musculus.bed > mus_musculus.chr.bed
|----Example for using
$ conda activate py3
$ infer_experiment.py -r /mnt/d/mm10_BedBigBed/mus_musculus.chr.bed -i SW3_d4_1_S45Aligned.sortedByCoord.out.chr.bam
```
### 01-08.Installation of rmats2sashimiplot
[Master repository in GitHub](https://github.com/Xinglab/rmats2sashimiplot)
```bash
$ conda activate py2
$ python setup.py install
```
### 01-09.Installation of Aspera connect
[Offical website](https://downloads.asperasoft.com/en/downloads/8?list)
```bash
$ ./ibmasperaconnect.sh
$ echo 'export PATH=/home/uekilab/.aspera/connect/bin:$PATH' >> ~/.bashrc
$ source ~/.bashrc
```
#### Aspera code example of download from European [Nucleotide Archive](https://www.ebi.ac.uk/ena)
```bash
$ ascp -v -QT -l 300m -P33001 -i ~/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR157/000/SRR1573670/SRR1573670.fastq.gz /mnt/d/; 
```
### 01-10.Installation of [subread](http://subread.sourceforge.net/)
```bash
$ echo 'export PATH=/mnt/c/subread/bin:$PATH' >> ~/.bashrc
$ source ~/.bashrc
```
#### Subread code example for featureCounts
check [here](http://bioinf.wehi.edu.au/featureCounts/) to set up correct parameters
```bash
$ featureCounts ./Sample_1.Aligned.out.bam -T 8 -s 2 -p -Q 10 -a ./Mus_musculus.GRCm38.96.chr.gtf -o ./Sample_1.counts
```
### 01-11.Installation of cufflinks
```bash
$ sudo apt-get install cufflinks 
```
### 01-12.Installation of stringtie
```bash
$ conda install -c bioconda stringtie
```
### 01-13.Installation of Hisat2
```bash
$ conda activate py3
$ conda install -c bioconda hisat2
$ conda update hisat2
```
#### Hisat2 code example for building index
```bash
$ cd /mnt/d/mm10_GTF
```
##### extract splice sites from a GTF file
```bash
$ hisat2_extract_splice_sites.py Mus_musculus.GRCm38.97.chr.gtf > mm10ss.txt
```
##### extract exon from a GTF file
```bash
$ hisat2_extract_exons.py Mus_musculus.GRCm38.97.chr.gtf > mm10ex.txt
```
##### Build index with ss (splice site) and exon information
```bash
$ mkdir /mnt/d/Hisat_Genome_Index_With_ss_exon
$ cd /mnt/d/Hisat_Genome_Index_With_ss_exon
$ hisat2-build -p $(grep -c ^processor /proc/cpuinfo) --ss /mnt/d/mm10_GTF/mm10.19.ss.txt --exon /mnt/d/mm10_GTF/mm10.19.ex.txt -f /mnt/d/mm10_FASTa/Mus_musculus.GRCm38.dna.chromosome.19.fa mm10.19.
```
### 01-14.Environment Checking
```bash
$ cufflinks -v
$ tophat2 --version
$ apt --version
$ conda --version
$ ascp --version
$ python --version
$ STAR --version
$ featureCounts -v
$ apt-cache policy python-htseq
$ bowtie2 --version
$ conda activate
$ source activate py2
$ pip --version
$ tophat --version
$ python --version
$ conda activate salmon
$ salmon --version
$ conda deactivate
$ conda deactivate
$ conda deactivate
```
## 02.Code example for bash (Unix/shell) language
### [02-01.Bash String Manipulation Examples – Length, Substring, Find and Replace](https://www.thegeekstuff.com/2010/07/bash-string-manipulation)
In bash shell, when you use a dollar sign followed by a variable name, shell expands the variable with its value. This feature of shell is called parameter expansion. But parameter expansion has numerous other forms which allow you to expand a parameter and modify the value or substitute other values in the expansion process. In this article, let us review how to use the parameter expansion concept for string manipulation operations. This article is part of the on-going bash tutorial series. Refer to our earlier article on bash { } expansion.
1. Identify String Length inside Bash Shell Script
```bash
${#string}
```
The above format is used to get the length of the given bash variable.
```bash
$ cat len.sh
#! /bin/bash
var="Welcome to the geekstuff"
echo ${#var}
$ ./len.sh
24
```
To understand more about bash variables, read 6 Practical Bash Global and Local Variable Examples.
2. Extract a Substring from a Variable inside Bash Shell Script
Bash provides a way to extract a substring from a string. The following example expains how to parse n characters starting from a particular position.
```bash
${string:position}
```
Extract substring from $string at $position
```bash
${string:position:length}
```
Extract $length of characters substring from $string starting from $position. In the below example, first echo statement returns the substring starting from 15th position. Second echo statement returns the 4 characters starting from 15th position. Length must be the number greater than or equal to zero.
```bash
$ cat substr.sh
#! /bin/bash
var="Welcome to the geekstuff"
echo ${var:15}
echo ${var:15:4}
$ ./substr.sh
geekstuff
geek
```
Also, refer to our earlier article to understand more about $*, $@, $#, $$, $!, $?, $-, $_ bash special parameters.
3. Shortest Substring Match
Following syntax deletes the shortest match of $substring from front of $string
```bash
${string#substring}
```
Following syntax deletes the shortest match of $substring from back of $string
```bash
${string%substring}
```
Following sample shell script explains the above two shortest substring match concepts.
```bash
$ cat shortest.sh
#! /bin/bash
filename="bash.string.txt"
echo ${filename#*.}
echo ${filename%.*}
$ ./shortest.sh
```bash
After deletion of shortest match from front: string.txt  
After deletion of shortest match from back: bash.string  
In the first echo statement substring ‘*.’ matches the characters and a dot, and # strips from the front of the string, so it strips the substring “bash.” from the variable called filename. In second echo statement substring ‘.*’ matches the substring starts with dot, and % strips from back of the string, so it deletes the substring ‘.txt’
4. Longest Substring Match
Following syntax deletes the longest match of $substring from front of $string
```bash
${string##substring}
```
Following syntax deletes the longest match of $substring from back of $string
```bash
${string%%substring}
```
Following sample shell script explains the above two longest substring match concepts.
```bash
$ cat longest.sh
#! /bin/bash
filename="bash.string.txt"
echo "After deletion of longest match from front:" ${filename##*.}
echo "After deletion of longest match from back:" ${filename%%.*}
$ ./longest.sh
```
After deletion of longest match from front: txt  
After deletion of longest match from back: bash  
In the above example, ##*. strips longest match for ‘*.’ which matches “bash.string.” so after striping this, it prints the remaining txt. And %%.* strips the longest match for .* from back which matches “.string.txt”, after striping  it returns “bash”.  
5. Find and Replace String Values inside Bash Shell Script
Replace only first match
```bash
${string/pattern/replacement}
```
It matches the pattern in the variable $string, and replace only the first match of the pattern with the replacement.
```bash
$ cat firstmatch.sh
#! /bin/bash
filename="bash.string.txt"
echo "After Replacement:" ${filename/str*./operations.}
$ ./firstmatch.sh
```
After Replacement: bash.operations.txt  
Replace all the matches  
```bash
${string//pattern/replacement}
```
It replaces all the matches of pattern with replacement.
```bash
$ cat allmatch.sh
#! /bin/bash
filename="Path of the bash is /bin/bash"
echo "After Replacement:" ${filename//bash/sh}
$ ./allmatch.sh
```
After Replacement: Path of the sh is '/bin/sh'  
Taking about find and replace, refer to our earlier articles – sed substitute examples and Vim find and replace.  
Replace beginning and end  
```bash
${string/#pattern/replacement}
```
Following syntax replaces with the replacement string, only when the pattern matches beginning of the $string.  
```bash
${string/%pattern/replacement}
```
Following syntax replaces with the replacement string, only when the pattern matches at the end of the given $string.
```bash
$ cat posmatch.sh
#! /bin/bash
filename="/root/admin/monitoring/process.sh"
echo "Replaced at the beginning:" ${filename/#\/root/\/tmp}
echo "Replaced at the end": ${filename/%.*/.ksh}
$ ./posmatch.sh
```
Replaced at the beginning: /tmp/admin/monitoring/process.sh  
Replaced at the end: /root/admin/monitoring/process.ksh  
### 02-02.Code example for [bash (Unix/shell) language](https://tiswww.case.edu/php/chet/bash/bashref.html) with loop
#### Unzip gz file
```bash
#!/bin/bash
number=$(ls | grep '\.gz$' | wc -l)
echo "This folder has ${number} gz files";
for i in $(seq 1 ${number})
do
  seq="NR==${i}";
  name=$(find . -maxdepth 1 -type f -name "*.gz" | awk ${seq})
  echo "Unzip ${name}";
  gunzip ${name}
done
```
#### Find out fq files within a directory (for single end)
```bash
#!/bin/bash
level=3
number=$(find . -maxdepth ${level} -type f -name "*.fq"| wc -l)
core=$(grep -c ^processor /proc/cpuinfo)
echo "This folder has ${number} fastq files";
for i in $(seq 1 ${number})
do
  seq1="NR==${i}";
  name1=$(find . -maxdepth ${level} -type f -name "*.fq" | awk ${seq1})
  name="./$(echo ${name1} | cut -d'/' -f 2).";
  echo "Mapping sample ${i} with ${core} CPU cores (${name}): ${name1}";
  STAR --runThreadN ${core} --alignEndsType EndToEnd --outSAMtype BAM SortedByCoordinate --outSAMmapqUnique 60 --outFileNamePrefix ${name} --genomeDir /mnt/d/GenomeIndex_ENSEMBL_STAR_ReadLength49 --readFilesIn ${name1}
done
```
#### Find out fq files within a directory (by ordering)
```bash
#!/bin/bash
number=$(ls | grep '\.fq$' | wc -l)
core=$(grep -c ^processor /proc/cpuinfo)
echo "This folder has ${number} fastq files";
for i in $(seq 1 2 $((${number}-1)))  
do
  seq1="NR==${i}";
  seq2="NR==$((${i}+1))";
  name1=$(find . -maxdepth 1 -type f -name "*.fq" | awk ${seq1})
  name2=$(find . -maxdepth 1 -type f -name "*.fq" | awk ${seq2})
  echo "Mapping sample $((${i}/2)) with ${core} CPU cores: ${name1} and ${name2}";
  STAR --runThreadN ${core} --outSAMtype BAM SortedByCoordinate --outSAMmapqUnique 60 --outFileNamePrefix ${name1} --genomeDir /mnt/d/GenomeIndex_ENSEMBL_STAR_ReadLength49 --readFilesIn ${name1} ${name2}
done
```
#### Find out fq files within a directory (by matching ordering)
```bash
#!/bin/bash
number=$(ls | grep '\.fq$' | wc -l)
core=$(grep -c ^processor /proc/cpuinfo)
echo "This folder has ${number} fastq files";
for i in $(seq 1 2 $((${number}-1)))  
do
  seq1="NR==${i}";
  seq2="NR==$((${i}+1))";
  name1=$(find . -maxdepth 1 -type f -name "*.fq" | awk ${seq1})
  name2=$(find . -maxdepth 1 -type f -name "*.fq" | awk ${seq2})
  echo "Mapping sample $((${i}+1)) with ${core} CPU cores (${name}): ${name1} and ${name2}";
  STAR --runThreadN ${core} --outSAMtype BAM SortedByCoordinate --outSAMmapqUnique 60 --outFileNamePrefix ${name} --genomeDir /mnt/d/GenomeIndex_ENSEMBL_STAR_ReadLength49 --readFilesIn ${name1} ${name2}
done
```
#### Find out fq files within a directory (by pre-defined array)
```bash
#!/bin/bash
files=(4 5 6 7 8 9 25 26 27 28 29 30)
number=$(ls | grep '\.fq$' | wc -l)
core=$(grep -c ^processor /proc/cpuinfo)
echo "This folder has ${number} fastq files";
for i in $(seq 0 $((${number}/2-1)))
do
  name="./S${files[${i}]}.";
  name1="./${files[${i}]}_1.fq";
  name2="./${files[${i}]}_2.fq";
  echo "Mapping sample $((${i}+1)) with ${core} CPU cores (${name}): ${name1} and ${name2}";
  STAR --runThreadN ${core} --outSAMtype BAM SortedByCoordinate --outSAMmapqUnique 60 --outFileNamePrefix ${name} --genomeDir /mnt/f/GenomeIndex_ENSEMBL_STAR_ReadLength99 --readFilesIn ${name1} ${name2}
done
```
